app/screens/select_server/select_server.js

```diff
@@ -324,6 +339,12 @@ export default class SelectServer extends PureComponent {
                 return;
             }
 
+            if (result.error && this.attemptWithProtocol === PROTOCOL.HTTPS) {
```

Comment:
What if instead of using a member variable to track whether or not to attempt (sorry, I'm apparently against extra state right now), we just pass it as a an argument to `pingServer`? Like
```javascript
handleClick = () => {
    this.pingServer(url, true);
}

pingServer = (url, retryWithHttp = false) => {
    getPing().then((result) => {
        if (result.error && retryWithHttp) {
            this.pingServer(url.replace(...), false);
        }
    }
}
```